File: algorithms/c/21.c
#include <stdio.h>
#include <stdlib.h>

struct ListNode
{
    int val;
    struct ListNode *next;
};

struct ListNode* mergeTwoLists(struct ListNode* l1, struct ListNode* l2)
{
    struct ListNode *p = (struct ListNode *)malloc(sizeof(struct ListNode));
    struct ListNode *result = p;
    while ( l1 && l2 )
    {
        p->next = (struct ListNode *)malloc(sizeof(struct ListNode));
        p = p->next;
        if ( l1->val < l2->val )
        {
            p->val = l1->val;
            l1 = l1->next;
        }
        else
        {
            p->val = l2->val;
            l2 = l2->next;
        }
    }
    while ( l1 )
    {
        p->next = (struct ListNode *)malloc(sizeof(struct ListNode));
        p = p->next;
        p->val = l1->val;
        l1 = l1->next;
    }
    while ( l2 )
    {
        p->next = (struct ListNode *)malloc(sizeof(struct ListNode));
        p = p->next;
        p->val = l2->val;
        l2 = l2->next;
    }
    p->next = NULL;
    p = result;
    result = result->next;
    free(p);
    return result;
}

int main()
{
    return 0;
}

File: algorithms/python/21.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def mergeTwoLists(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        if not l1: return l2
        elif not l2: return l1
        if l1.val > l2.val:
            l1, l2 = l2, l1
        p = l1
        while l2:
            while p.next and p.next.val < l2.val:
                p = p.next
            node, l2 = l2, l2.next
            p.next, node.next = node, p.next
        return l1
